DocsNav/scripts/insert_navbar.jl

```diff
@@ -0,0 +1,165 @@
+#   - Inserts the new navbar block immediately after the first <body> tag.
+#   - If the fetched navbar content does not contain the markers, it is wrapped (without further modification) with:
+#         <!-- NAVBAR START -->
+#         (navbar content)
+#         <!-- NAVBAR END -->
+#   - If a file does not contain a <body> tag, that file is skipped.
+#   - An optional --exclude parameter (comma‑separated) will skip any file whose path contains one of the provided substrings.
+#
+# Required package: HTTP
+#
+# (Install HTTP via Julia’s package manager, e.g. using Pkg; Pkg.add("HTTP"))
+
+using HTTP
+
+# --- Utility: Read file contents ---
+function read_file(filename::String)
+    open(filename, "r") do io
+        read(io, String)
+    end
+end
+
+# --- Utility: Write contents to a file ---
+function write_file(filename::String, contents::String)
+    open(filename, "w") do io
+        write(io, contents)
+    end
+end
+
+# --- Exclusion Function ---
+function should_exclude(filename::String, patterns::Vector{String})
+    for pat in patterns
+        if occursin(pat, filename)
+            return true
+        end
+    end
+    return false
+end
+
+# --- Remove any existing navbar block and any whitespace following it ---
+function remove_existing_navbar(html::String)
+    start_marker = "<!-- NAVBAR START -->"
+    end_marker   = "<!-- NAVBAR END -->"
+    while occursin(start_marker, html) && occursin(end_marker, html)
+        start_idx_range = findfirst(start_marker, html)
+        end_idx_range   = findfirst(end_marker, html)
+        # Extract the first index from the returned range.
+        start_idx = first(start_idx_range)
+        end_idx   = first(end_idx_range)
+        # Get prefix: everything before the start marker (or empty if start_idx is 1)
+        prefix = start_idx > 1 ? html[1:start_idx-1] : ""
+        # Suffix: from the end of the end marker to the end of the file,
+        # then remove any leading whitespace.
+        suffix = lstrip(html[end_idx + length(end_marker) : end])
+        html = string(prefix, suffix)
+    end
+    return html
+end
+
+# --- Wrap navbar HTML with markers if not already present ---
+function wrap_navbar(navbar_html::String)
+    if !occursin("NAVBAR START", navbar_html) || !occursin("NAVBAR END", navbar_html)
+        return "<!-- NAVBAR START -->\n" * navbar_html * "\n<!-- NAVBAR END -->"
+    else
+        return navbar_html
+    end
+end
+
+# --- Insert new navbar into HTML ---
+function insert_navbar(html::String, navbar_html::String)
+    # Remove any previously inserted navbar block (and any trailing whitespace).
+    html = remove_existing_navbar(html)
+    # Use a regex to find the first <body> tag (case-insensitive).
+    m = match(r"(?i)(<body[^>]*>)", html)
+    if m === nothing
+        println("Warning: Could not find <body> tag in the file; skipping insertion.")
+        return html  # Return the unmodified HTML.
+    end
```

Comment:
I think we should pass the filename to this function and do:

```suggestion
    if m === nothing
        @warn "Could not find <body> tag in the file $(filename); skipping insertion."
        return html  # Return the unmodified HTML.
    end
```